lib/resque/worker.rb

```diff
@@ -503,17 +503,15 @@ def heartbeat!(time = Time.now)
     end
 
     def start_heartbeat
-      heartbeat!
```

Comment:
because pruning a worker that is still running a job can cause it to be re-enqueued for a retry while still running. So the job could be running twice concurrently